src/bin/cargo/commands/owner.rs

```diff
@@ -33,19 +82,70 @@ pub fn cli() -> Command {
         ))
 }
 
+fn add_registry_args() -> [Arg; 3] {
+    [
+        opt("index", "Registry index URL to modify owners for")
+            .value_name("INDEX")
+            .conflicts_with("registry"),
+        opt("registry", "Registry to modify owners for").value_name("REGISTRY"),
+        opt("token", "API token to use when authenticating").value_name("TOKEN"),
+    ]
+}
+
 pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
+    let (to_add, to_remove, list) = match args.subcommand() {
+        Some(("add", subargs)) => (
+            subargs
+                .get_many::<String>("add")
+                .map(|xs| xs.cloned().collect::<Vec<String>>()),
+            None,
+            false,
+        ),
+        Some(("remove", subargs)) => (
+            None,
+            subargs
+                .get_many::<String>("remove")
+                .map(|xs| xs.cloned().collect()),
+            false,
+        ),
+        Some(("list", _)) => (None, None, true),
+        _ => (
+            args.get_many::<String>("add")
+                .map(|xs| xs.cloned().collect::<Vec<String>>()),
+            args.get_many::<String>("remove")
+                .map(|xs| xs.cloned().collect()),
+            args.flag("list"),
+        ),
```

Comment:
I assume this is meant to operate for `None`.  We should separate out `Some(_)` and panic for it as we should only get the subcommands back we defined.

Reply:
If there are no subcommands or any parameters(like：```--add```，..), an error message will be displayed instead of panic.

https://github.com/rust-lang/cargo/pull/11879/files#diff-981bc8fd9f47eff85de2c539a5be58b0002e2f060064a090218060c5ae3e5722R114-R122